qcodes/instrument_drivers/QDevil/QDevil_QDAC.py

```diff
@@ -700,7 +712,7 @@ def _setslope(self, chan, slope):
             try:
                 fg = self._assigned_fgs[chan]
                 self._assigned_fgs[chan].t_end = 0
-                self._assigned_triggers.pop(fg)
+                self._assigned_triggers.pop(fg.fg)
```

Comment:
This is a change but as far as I can see _assigned_triggers does not contain Generator objects but integers so this code has probably never worked correctly

Reply:
AFAICS, `fg` is an integer (representing a function generator), so `fg.fg` does not make sense, and the original `pop(fg)` seems right.

Reply:
If you look at line 882 you can see that the variables in self._assigned_fgs are instances of the Generator Class and not integers. 
As far as I can see you therefore need to do `fg.fg` to get the generator number which is what is used within `self._assigned_triggers` 

See 
 https://github.com/QCoDeS/Qcodes/blob/0813f163cb3dbf5ffbbff9bc64edf235c13a1f5d/qcodes/instrument_drivers/QDevil/QDevil_QDAC.py#L882

Reply:
Yes, you're right.  I guess I should work on making the code easier to understand :-/

Reply:
Yes that would be great. These dictionaries are a bit confusing in their types